plugins/europeana/search.js

```diff
@@ -75,15 +75,22 @@ export function rangeFromQueryParam(paramValue) {
  */
 function resultsFromApiResponse(response) {
   const items = response.data.items;
+  const hits = response.data.hits;
 
   const results = items.map(item => {
+    let selector;
+    if (hits) {
+      selector = hits.find((hit) => item.id === hit.scope);
+    }
+
     return {
       europeanaId: item.id,
       edmPreview: item.edmPreview ? `${item.edmPreview[0]}&size=w200` : genericThumbnail(item.id, { type: item.type, size: 'w200' }),
       dcTitle: item.dcTitleLangAware,
       dcDescription: item.dcDescriptionLangAware,
       dcCreator: item.dcCreatorLangAware,
-      edmDataProvider: item.dataProvider
+      edmDataProvider: item.dataProvider,
+      selector: (selector && selector.selectors) ? selector.selectors : []
```

Comment:
I think something like this will be better:
```
/**
 * Extract search results from API response
 * @param  {Object} response API response
 * @return {Object[]} search results
 */
function resultsFromApiResponse(response) {
  const items = response.data.items;

  const results = items.map(item => {
    return {
      ...{
        europeanaId: item.id,
        edmPreview: item.edmPreview ? `${item.edmPreview[0]}&size=w200` : genericThumbnail(item.id, {
          type: item.type,
          size: 'w200'
        }),
        dcTitle: item.dcTitleLangAware,
        dcDescription: item.dcDescriptionLangAware,
        dcCreator: item.dcCreatorLangAware,
        edmDataProvider: item.dataProvider
      },
      ...(response.data.hits !== undefined ? hitForItem(response.data.hits, item.id) : {})
    };
  });

  return results;
}

function hitForItem(hits, id) {
  const selector = hits.find((hit) => id === hit.scope);
  return selector ? { selector: selector.selectors[0]  } : {};
}
```
Then in components/search/SearchResults.vue, we can just have this:
```
-          if (result.selector.length === 0 && result.dcDescription) texts.unshift(result.dcDescription);
+          if (!result.selector && result.dcDescription) texts.unshift(result.dcDescription);
```
 
And from a data point of view only items with hits will actually pass a selector object along.

Reply:
Don't you think this is harder to read? Also, it seems like we are adding more code. I am happy to pass an object instead of an array though.

Reply:
It's more consistent with what we do elsewhere where objects are constructed.
And the `hitForItem` function is more atomic, being removed from the `resultsFromApiResponse` function.

But the main reason is to not have the key set to an empty value, but rather not added at all.
The other way to achieve this would be to remove any keys with blank values, but again the approach of appending the additional key like this is how we do it elsewhere.